server/py/framework/db/sqldb/db.py

```diff
@@ -3113,17 +3113,69 @@ def skip_deletion():
                 project=project,
             )
 
-        query = session.query(main_table).filter(where_clause)
-        deletions_count = query.delete(synchronize_session=False)
-        log_kwargs = {
-            "deletions_count": deletions_count,
-            "main_table": main_table,
-            "project": project,
-            "main_table_identifier": main_table_identifier,
-        }
-        logger.debug("Removed rows from table", **log_kwargs)
-        session.commit()
-        return deletions_count
+        total_deleted = SQLDB._delete_table_in_batches(
+            session, main_table, where_clause
+        )
+        # redundant log ?
```

Comment:
Not redundant because you want to know that the operation has ended (replace with "info")